components/patient-data/impl/src/test/java/org/phenotips/data/internal/controller/GeneListControllerTest.java

```diff
@@ -0,0 +1,296 @@
+import org.phenotips.data.PatientData;
+import org.phenotips.data.PatientDataController;
+import org.xwiki.bridge.DocumentAccessBridge;
+import org.xwiki.component.manager.ComponentLookupException;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.test.mockito.MockitoComponentMockingRule;
+
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import org.hamcrest.Matchers;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xpn.xwiki.objects.BaseStringProperty;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Test for the {@link GeneListController} Component,
+ * only the overridden methods from {@link AbstractComplexController} are tested here
+ */
+public class GeneListControllerTest
+{
+
+    @Rule
+    public MockitoComponentMockingRule<PatientDataController<Map<String, String>>> mocker =
+        new MockitoComponentMockingRule<PatientDataController<Map<String, String>>>(GeneListController.class);
+
+    private static final String GENES_STRING = "genes";
+
+    private static final String CONTROLLER_NAME = GENES_STRING;
+
+    private static final String GENES_ENABLING_FIELD_NAME = GENES_STRING;
+
+    private static final String GENES_COMMENTS_ENABLING_FIELD_NAME = "genes_comments";
+
+    private static final String GENE_KEY = "gene";
+
+    private static final String COMMENTS_KEY = "comments";
+
+    private DocumentAccessBridge documentAccessBridge;
+
+    private List<BaseObject> geneXWikiObjects;
+
+    @Mock
+    private Patient patient;
+
+    @Mock
+    private XWikiDocument doc;
+
+    @Mock
+    private BaseObject obj1;
+
+    @Mock
+    private BaseObject obj2;
+
+    @Mock
+    private BaseObject obj3;
+
+
+    @Before
+    public void setUp() throws Exception
+    {
+        MockitoAnnotations.initMocks(this);
+
+        this.documentAccessBridge = this.mocker.getInstance(DocumentAccessBridge.class);
+
+        DocumentReference patientDocument = new DocumentReference("wiki", "patient", "00000001");
+        doReturn(patientDocument).when(this.patient).getDocument();
+        doReturn(this.doc).when(this.documentAccessBridge).getDocument(patientDocument);
+        this.geneXWikiObjects = new LinkedList<>();
+        this.geneXWikiObjects.add(obj1);
+        this.geneXWikiObjects.add(obj2);
+        this.geneXWikiObjects.add(obj3);
+        int i = 1;
+        for(BaseObject gene : this.geneXWikiObjects){
```

Comment:
Does this pass style checker? I thought there should be a space before `(` and after `)`?